cluster_pack/packaging.py

```diff
@@ -368,19 +369,40 @@ def detect_archive_names(
         pex_file = ""
         env_name = packer.env_name()
 
+    def build_package_path(name: str = env_name,
+                           extension: Optional[str] = packer.extension()) -> str:
+        path = (f"{get_default_fs()}/user/{getpass.getuser()}"
+                f"/envs/{name}")
+        if extension is None:
+            return path
+        return f"{path}.{extension}"
+
     if not package_path:

```

Comment:
if package_path is provided I feel like it can be changed in other if, is it expected ?

Reply:
yes, if package_path is provided, we can add a .zip at the end if large_pex is true. This was the previous behavior, the code_path that is added here is for the case when large_pex is None but we detect we are running from unzipped, and we don't change the previous behavior